syncmymoodle/__main__.py

```diff
@@ -178,8 +178,13 @@ def __init__(self, config):
 
     def login(self):
         def get_session_key(soup):
-            session_key = soup.find("a", {"data-title": "logout,moodle"})["href"]
-            return re.findall("sesskey=([a-zA-Z0-9]*)", session_key)[0]
+            try:
+                session_key = soup.find("input", {"name": "sesskey"})["value"]
+            except TypeError:
+                logger.critical("Can't retrieve session key")
+                logger.info(soup)
+                exit(1)
```

Comment:
`exit` is intended for interactive shells, in scripts `sys.exit` should be used. Raising an exception would also be fine by me.

Reply:
Oh, true that.

Reply:
Fair enough. Although I just copied this verbatim (exit(1) is used in 5 other places in __main__.py)

Reply:
Note: We actually use `exit()` in several different places.

https://github.com/Romern/syncMyMoodle/blob/9727e0b8f7d94b3586b015ebc377d0f6adaea457/syncmymoodle/__main__.py#L216

https://github.com/Romern/syncMyMoodle/blob/9727e0b8f7d94b3586b015ebc377d0f6adaea457/syncmymoodle/__main__.py#L323

https://github.com/Romern/syncMyMoodle/blob/9727e0b8f7d94b3586b015ebc377d0f6adaea457/syncmymoodle/__main__.py#L1204

https://github.com/Romern/syncMyMoodle/blob/9727e0b8f7d94b3586b015ebc377d0f6adaea457/syncmymoodle/__main__.py#L1191

https://github.com/Romern/syncMyMoodle/blob/9727e0b8f7d94b3586b015ebc377d0f6adaea457/syncmymoodle/__main__.py#L1231

Reply:
Okay, then just leave it. Better to tackle this in a separate issue :D